Lib/test/test_buffer.py

```diff
@@ -4412,6 +4412,24 @@ def test_issue_7385(self):
         x = ndarray([1,2,3], shape=[3], flags=ND_GETBUF_FAIL)
         self.assertRaises(BufferError, memoryview, x)
 
+    # Test PyBuffer_SizeFromFormat()
+    @support.cpython_only
+    def test_pybuffer_size_from_format(self):
+        from ctypes import pythonapi
+        PyBuffer_SizeFromFormat = pythonapi.PyBuffer_SizeFromFormat
+
+        # basic tests
+        self.assertIs(PyBuffer_SizeFromFormat("3si") > 0 , True)
```

Comment:
I suggest to test a few formats using a loop and compare the result to struct.calcsize(). Something like:

```
for format in ("i", "3s", "0i", ""):
    self.assertEqual(PyBuffer_SizeFromFormat(format.encode()), struct.calcsize(format))
```

You must explicitly encode the Unicode format string to UTF-8 to pass it as a byte string (char*), no?